### PR TITLE
Scripts and README for verification

### PR DESCRIPTION
**Pull request by bsubercaseaux**
The README now includes information about verification. Namely, how to verify the "tautology proof" (e.g., the cube split is exhaustive) and how to (partially) verify the "UNSAT proof" by providing a simple way of checking that cube $i$ is UNSAT. As cubes take a few seconds to run, a dedicated reviewer can easily check thousands of them.

The instructions also include the use of `cake_lpr`, the verified LRAT checker.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,8 +54,47 @@ the CNFs produced can be shown UNSAT instantly.
...
+```
+
+For the **UNSAT proof**, given the total amount of computation required to run all cubes is probably unfeasible for any verifier, we provide a simple way to verify any given cube to be UNSAT. To generate the formula + cube number $i$ (assuming the steps for the **tautology proof** have been followed), run:
+```
+./cube_gen <i> vars.out > .tmp_cube_<i>
+python3 cube_join.py -f 6-30.cnf -c .tmp_cube_<i> -o with_cube_<i>.cnf
+cadical with_cube_<i>.cnf proof_cube_<i>.lrat --lrat
+cake_lpr with_cube_<i>.cnf proof_cube_<i>.lrat
+```
+To simplify the process, we have added a bash script `solve_cube.sh`, which can be simply run as (for example, for cube number 42):
```
Comment (JamesGallicchio) [inline README.md]: Maybe we should add a script for the tautology proof as well?

Comment (JamesGallicchio) [reply README.md]: oh wait there is such a script

File: cube_checking/cube_gen
Comment (JamesGallicchio) [inline cube_checking/cube_gen:1]: should this be committed?

Comment (bsubercaseaux) [reply cube_checking/cube_gen:1]: oops, definitely not, it's OS dependent. My bad!

File: cube_checking/cubes.icnf
Comment (JamesGallicchio) [inline cube_checking/cubes.icnf:1]: again I think this should be removed from the commit (maybe added to gitignore as well?)

File: README.md (hunk 1 above)
Comment (bsubercaseaux) [reply README.md]: Oops, the tautology script was accidentally pushed. I will add a corrected version.
The reason why I wanted to make the reviewer go command by command is to see what's happening (?)

File: cube_checking/cubes.icnf
Comment (bsubercaseaux) [reply cube_checking/cubes.icnf:1]: yep, sorry, I was incorrectly assuming that this thing would be in the git ignore

File: cube_checking/cube_gen
Comment (bsubercaseaux) [reply cube_checking/cube_gen:1]: Removed :)

File: cube_checking/cubes.icnf
Comment (bsubercaseaux) [reply cube_checking/cubes.icnf:1]: Removed :)

File: README.md (hunk 1 above)
Comment (JamesGallicchio) [reply README.md]: Yeah. I think it's fine to put it in a script if the script is straightforward.

Comment (bsubercaseaux) [reply README.md]: Added this :)

File: README.md [hunk 2]
```diff
@@ -54,8 +54,50 @@ the CNFs produced can be shown UNSAT instantly.
...
+In order to verify that the resulting formula $F$ is indeed UNSAT, two things ought to be checked:
+
+ 1) **(Tautology proof)** The formula $F$ is UNSAT iff $F \land C_i$ is UNSAT for every $i$.
+ 2) **(UNSAT proof)** For each cube $C_i$, the formula $F \land C_i$ is UNSAT.
+
+The **tautology proof** is implied by checking that $F \land (\bigwedge_i \neg C_i)$ is UNSAT, as per Section 7.3 of [Heule and Scheucher](https://arxiv.org/abs/2403.00737).
+
+As a first step, we will generate the main CNF by running:
+```
+lake exe encode hole 6 30 cube_checking/vars.out > cube_checking/6-30.cnf
```
Comment (JamesGallicchio) [inline README.md]: At least on my computer this needs to be run within the `Lean/` folder.
```
cd Lean
lake exe encode hole 6 30 ../cube_checking/vars.out > ../cube_checking/6-30.cnf
```

**Review by JamesGallicchio — CHANGES_REQUESTED** [review]
I'm waiting for everything to compile on my computer but otherwise looks good to me!

File: README.md (hunk 2 above)
Comment (bsubercaseaux) [reply README.md]: good catch, updated!

Comment (ccodel) [reply README.md]: This comment change should probably be made to the "Generating CNFs" section in the README as well. I'll add a commit for this.

Comment (bsubercaseaux) [reply README.md]: Done!

**Review by JamesGallicchio — APPROVED** [review]

**Pull request merged**